core/sdk-core/src/main/java/software/amazon/awssdk/core/rules/FnNode.java

```diff
@@ -67,6 +67,10 @@ public Fn validate() {
                 return new ParseUrl(this);
             case Substring.ID:
                 return new Substring(this);
+            case UriEncodeFn.ID:
+                return new UriEncodeFn(this);
+            case IsVirtualHostableS3Bucket.ID:
```

Comment:
Is there any future plans to separate out service specific rules?

Reply:
Not how how that would be possible since the rules engine needs to be aware of it; there's no way to define and inject external rules at the moment.